agent/pulp/agent/gofer/pulpplugin.py

```diff
@@ -118,6 +151,35 @@ def get_secret():
     return bundle.uid()
 
 
+class Attach(Thread):
+    """
+    This thread (task) persistently:
+      - validates the registration status
+      - if registered, updates the plugin settings and attach.
+      - if not registered, detach the plugin.
+    The reason for doing this in a thread is that we don't
+    want to block in the initializer.
+    """
+    
+    def __init__(self):
+        super(Attach, self).__init__()
+        self.setDaemon(True)
+    
+    def run(self):
+        while True:
+            try:
+                validate_registration()
+                if registered:
+                    update_settings()
+                    plugin.attach()
+                else:
+                    plugin.detach()
+                break
```

Comment:
maybe add a comment here to make it more obvious that this breaks out of the "while True" loop. It took me some time to figure this out when reviewing :cat2: